sceptre/scipool/templates/sc-enduser-iam.yaml

```diff
@@ -66,7 +66,10 @@ Resources:
           - Effect: Allow
             Action:
               - ssm:StartSession
-            Resource: !Sub "arn:aws:ec2:${AWS::Region}:${AWS::AccountId}:instance/*"
+            Resource:
+              - !Sub "arn:aws:ssm:*:${AWS::AccountId}:document/AWS-StartSSHSession"
+              - !Sub "arn:aws:ssm:*:${AWS::AccountId}:document/AWS-StartInteractiveCommand"
+              - !Sub "arn:aws:ec2:${AWS::Region}:${AWS::AccountId}:instance/*"
```

Comment:
The "*" at the end of the ARN pattern makes me wonder what keeps one user from accessing another user's machine.

Reply:
the restriction is here: https://github.com/Sage-Bionetworks-IT/organizations-infra/blob/master/sceptre/scipool/templates/sc-enduser-iam.yaml

that works because the ssm session is tagged with the synapse user and resolves to the aws userid, which is the same as a tag on the ec2

Reply:
There's a conditional statement below that limits the allowing start-session to 'callers' match a tag on the instance: